File: CS50x/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int FindLetters(string paragraph);
int FindSentences(string paragraph);
int FindWords(string paragraph);

int main(void)
{
    // Get story text from user
    string story = get_string("Text: ");

    // Calculate words sentences and letters
    int sentences = FindSentences(story);
    int words = FindWords(story);
    int letters = FindLetters(story);

    // Perform grade calculation
    double wordsPerHundredMulti = (double)100 / (double)words;
    double L = letters * wordsPerHundredMulti;
    double S = sentences * wordsPerHundredMulti;
    double indexDouble = 0.0588 * L - 0.296 * S - 15.8;
    int index = round(indexDouble);

    // Output the correct grade to the screen
    if (index >= 16) printf("Grade 16+\n");
    else if (index >= 2 && index <= 15) printf("Grade %i\n", index);
    else printf("Before Grade 1\n");


}

// Loop to find all sentences
int FindSentences(string paragraph)
{
    int sentences = 0;
    for (int i = strlen(paragraph); i >= 0; i--)
    {
        if (paragraph[i] == 46 || paragraph[i] == 33 || paragraph[i] == 63) sentences++;
    }
    return sentences;
}

// Function to find how many letters
int FindLetters(string paragraph)
{
    int letters = 0;
    for (int i = strlen(paragraph); i >= 0; i--)
    {
        if (toupper(paragraph[i]) >= 65 && toupper(paragraph[i]) <= 90) letters++;
    }
    return letters;
}

// Find how many words there are
int FindWords(string paragraph)
{
    int words = 0;
    for (int i = strlen(paragraph); i >= 0; i--)
    {
        if (paragraph[i] == 32) words++;
    }
    return words + 1; // Need +1 because we calculate the amount of words by spaces and the first word doesn't have a space
}

File: CS50x/pset6/readability/readability.py
def main():
    paragraph = input("Text: ")
    sentences = find_sentences(paragraph)
    words = find_words(paragraph)
    letters = find_letters(paragraph)

    # find the average words per 100
    words_per_hundred_multi = 100 / words
    # algorithim to find the grade index
    l = letters * words_per_hundred_multi
    s = sentences * words_per_hundred_multi
    index = round(0.0588 * l - 0.296 * s - 15.8)

    # cut off for grade 16+ and lower than grade 1
    if index >= 16:
        print("Grade 16+")
    elif index >= 2 and index <= 15:
        print(f"Grade {index}")
    else:
        print("Before Grade 1\n")

# loop through the paragraph and look for ., ? and !


def find_sentences(paragraph):
    sentences = 0
    for char in paragraph:
        if ord(char) == 46 or ord(char) == 33 or ord(char) == 63:
            sentences += 1
    return sentences

# loop through the paragraph and look for spaces, we add 1 extra word at the
# start because the first word doesn't have a space


def find_words(paragraph):
    words = 1
    for char in paragraph:
        if ord(char) == 32:
            words += 1
    return words

# loop through the paragraph and look any letter in the alphabet using ascii
# ord() returns the ascii number and we upp the letter so that weonly search
# the uppercase alphabet


def find_letters(paragraph):
    letters = 0
    for char in paragraph:
        if ord(char.upper()) >= 65 and ord(char.upper()) <= 90:
            letters += 1
    return letters


main()
